src/Reflection/ParameterReflection.php

```diff
@@ -92,10 +93,19 @@ public function detectType()
             return null;
         }
 
-        $params = $docBlock->getTags('param');
+        /** @var ParamTag[] $params */
+        $params       = $docBlock->getTags('param');
+        $paramTag     = $params[$this->getPosition()] ?? null;
```

Comment:
I don't fully understand this particular line: what is going on here? 👀

Reply:
Here we're trying to get the parameter from phpdoc block by position (the original implementation worked like that).
This is equivalent of `$paramTag = isset($params[$this->getPosition()]) ? $params[$this->getPosition()] : null;`

In case if we weren't able to find it - we're trying to get the type by the variable name.